package/sbp_rtcm3_bridge/src/sbp_rtcm3_bridge.c

```diff
@@ -152,62 +155,91 @@ static void utc_time_callback(u16 sender_id, u8 len, u8 msg[], void *context)
   rtcm2sbp_set_leap_second(leap_second,&state);
 }
 
+static int notify_simulator_enable_changed(void *context)
+{
+  (void)context;
+  sbp_simulator_enabled_set(simulator_enabled_watch);
+  return 0;
+}
+
 int main(int argc, char *argv[])
 {
+  int status = EXIT_SUCCESS;
+  settings_ctx_t *settings_ctx = NULL;
+  sbp_zmq_pubsub_ctx_t *ctx = NULL;
+
   logging_init(PROGRAM_NAME);
 
   if (parse_options(argc, argv) != 0) {
     piksi_log(LOG_ERR, "invalid arguments");
     usage(argv[0]);
-    exit(EXIT_FAILURE);
-  }
-
-  /* Need to init state variable before we get SBP in */
-  rtcm2sbp_init(&state, sbp_message_send, sbp_base_obs_invalid);
-
-  /* Prevent czmq from catching signals */
-  zsys_handler_set(NULL);
-
-  sbp_zmq_pubsub_ctx_t *ctx = sbp_zmq_pubsub_create(SBP_PUB_ENDPOINT,
-                                                    SBP_SUB_ENDPOINT);
-  if (ctx == NULL) {
-    exit(EXIT_FAILURE);
-  }
-
-  zsock_t *rtcm3_sub = zsock_new_sub(RTCM3_SUB_ENDPOINT, "");
-  if (rtcm3_sub == NULL) {
-    piksi_log(LOG_ERR, "error creating SUB socket");
-    exit(EXIT_FAILURE);
-  }
-
-  if (zloop_reader(sbp_zmq_pubsub_zloop_get(ctx), rtcm3_sub,
-                   rtcm3_reader_handler, NULL) != 0) {
-    piksi_log(LOG_ERR, "error adding reader");
-    exit(EXIT_FAILURE);
-  }
-
-  if (sbp_init(sbp_zmq_pubsub_rx_ctx_get(ctx),
-               sbp_zmq_pubsub_tx_ctx_get(ctx)) != 0) {
-    piksi_log(LOG_ERR, "error initializing SBP");
-    exit(EXIT_FAILURE);
-  }
-
-  if (sbp_callback_register(SBP_MSG_GPS_TIME, gps_time_callback, NULL) != 0) {
-    piksi_log(LOG_ERR, "error setting GPS TIME callback");
-    exit(EXIT_FAILURE);
-  }
+    status = EXIT_FAILURE;
+  } else {
+    /* Need to init state variable before we get SBP in */
+    rtcm2sbp_init(&state, sbp_message_send, sbp_base_obs_invalid);
 
-  if (sbp_callback_register(SBP_MSG_UTC_TIME, utc_time_callback, NULL) != 0) {
-    piksi_log(LOG_ERR, "error setting UTC TIME callback");
-    exit(EXIT_FAILURE);
-  }
+    /* Prevent czmq from catching signals */
+    zsys_handler_set(NULL);
 
-  if (sbp_callback_register(SBP_MSG_SETTINGS_READ_RESP, sbp_read_resp_callback, NULL) != 0) {
-    piksi_log(LOG_ERR, "error setting SBP READ RESP callback");
-    exit(EXIT_FAILURE);
+    ctx = sbp_zmq_pubsub_create(SBP_PUB_ENDPOINT, SBP_SUB_ENDPOINT);
```

Comment:
It looks like we went full ASIL style for this code now?  What's the justification?  Personally the old style (flat vs nested) was much easier to read.  I'm fine with replacing `exit(code)` with a `return code;` though.

Reply:
Happy to revert!